### ISSUE TITLE
tests

### ISSUE THREAD

**Issue by lekoala** [olifolkerd/tabulator#3754]
*Is your feature request related to a problem? Please describe.**
I just sent you 3 PR. They are fairly small and tested, but since the scope of this project is very big and there are many features, i can see how easy it would be to break things.

**Describe the solution you'd like**
did you consider adding automated tests that could bring some reassurance that PR don't break everything?
i've been using https://github.com/avajs/ava with some success, it works quite well and it's easy to have a github action that runs the test automatically
if that's something you would consider and/or are interested, I'd be happy to help set up this if I need to make some larger PR.

**Describe alternatives you've considered**
testing manually, but it's really tedious ;)

**Comment by olifolkerd** [comment]
Hey @lekoala 

I would be very interested in setting up some automated testing, i have put it out to the community, but havnt had any interest so far.

If this is something you would be interested in adding then that would be great.

Could you explain a bit more about your proposed approach. i would want to start small and allow tests to be added over time rather than building a massive monolithic set of tests that i then have to spend a lot of time supporting if there are issues.

Cheers

Oli :)

**Comment by lekoala** [comment]
@olifolkerd fully agreed, i don't have time myself to write all tests anyway, but at least if there is an issue, you can create the failing test, fix the issue, and see that the test pass. at least that's some starting point.

i can offer to do the same thing i did for this project https://github.com/lekoala/bootstrap5-tags
setup ava, a create one test to show how it works.
after that, it's up to anyone making a pr to fix a bug to create the corresponding test

**Comment by olifolkerd** [comment]
Sounds good

**Comment by ortwic** [comment]
Yes, automated testing is a very good thing. Ask uncle bob aka Robert C. Martin. :-) Hopefully the PR makes it soon into main.
